### ISSUE TITLE
shellcheck doesn't process bash's ${1:-default} correctly

### ISSUE THREAD

**Issue by jwaggy** [koalaman/shellcheck#1487]
#### For bugs
- SC2120 
- shellcheck version 0.6.0
- The rule's wiki page does not already cover this
- I tried on shellcheck.net and verified that this is still a problem on the latest commit

- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
lol()
{
echo "${1:-10}
}
lol

```

#### Here's what shellcheck currently says:
lol references arguments but none are ever passed.

#### Here's what I wanted or expected to see:
I would expect it to not throw an error/warning because ${1:-10} makes the variable default to "10" if $1 is unset and zero length, aka no argument is given, otherwise it is whatever is passed into $1.

**Comment by pawamoy** [comment]
Just a note: `:-` uses default when variable is empty, even if it is set:

```console
$ a=""
$ echo "${a:-10}"
10
```

**Comment by jwaggy** [comment]
I should have said unset 'or' zero length. Apologies.
